src/db/models/budgetSegmentBreakdown.ts

```diff
@@ -13,3 +15,61 @@ export const BUDGET_SEGMENT_BREAKDOWN_ID = brandedType<
   number,
   BudgetSegmentBreakdownId
 >(t.number);
+
+const BUDGET_SEGMENT_BREAKDOWN_FIXED_TYPE = t.keyof({
+  c: null,
+  p: null,
+});
+
+export const BUDGET_SEGMENT_BREAKDOWN_CONTENT = t.intersection([
+  t.type({
+    /**
+     * Some breakdowns seem to incorrectly use non-numeric values for amount
+     *
+     * TODO: Be more strict for future additions,
+     * clean old values and drop string type from this union
+     */
+    amount: t.union([t.number, t.string, t.null]),
+  }),
+  t.partial({
+    originalAmount: t.number,
+    /**
+     * Some percent figures are incorrectly using numbers as strings
+     *
+     * TODO: Introduce more strict checks and drop string type
+     */
+    percent: t.union([t.number, t.string]),
+    fixed: BUDGET_SEGMENT_BREAKDOWN_FIXED_TYPE,
+    breakdown: t.array(
+      t.intersection([
+        t.type({
+          amount: t.number,
+          percent: t.number,
+          description: t.string,
+        }),
+        t.partial({
+          fixed: BUDGET_SEGMENT_BREAKDOWN_FIXED_TYPE,
+        }),
+      ])
+    ),
+  }),
```

Comment:
I need a second opinion on this part from @s0 otherwise, everything looks good.

Reply:
These were extracted based on existing data that we already have in the database. Since we do v[alidation when reading records](https://github.com/UN-OCHA/hpc-api-core/blob/33e445cbf8794bb062f18ee65fd6b43e505538ee/src/db/util/raw-model.ts#L123), I defined everything that budget segment breakdown content JSON data can have.

Now, since JSON data was not validated in the past, there is garbage data in many cases, but I tried capturing all possibilities for the sake of reading from DB. There is an idea to split into the codecs for reading existing data and codecs creating new records, where we want to be very strict going forward.